consensus/src/messages/handlers.rs

```diff
@@ -593,20 +616,28 @@ impl<N: Network> Handle<N, Arc<RwLock<Blockchain>>> for RequestTransactionsProof
         _peer_id: N::PeerId,
         blockchain: &Arc<RwLock<Blockchain>>,
     ) -> Result<ResponseTransactionsProof, ResponseTransactionProofError> {
+        // Validate request.
+        if self.hashes.len() > Self::MAX_TRANSACTIONS {
+            return Err(ResponseTransactionProofError::TooManyTransactionsProvided);
+        }
+
         if self.hashes.is_empty() {
             // If we are not given a list of transactions then there is nothing to do
             return Err(ResponseTransactionProofError::NoTransactionsProvided);
         }
 
         // Handle the different cases: if we are provided a block number (to generate the proof) or not
         if let Some(block_number) = self.block_number {
-            prove_txns_with_block_number(blockchain, &self.hashes, block_number)
+            Self::prove_txns_with_block_number(blockchain, &self.hashes, block_number)
         } else {
-            prove_transaction(blockchain, &self.hashes[0])
+            Self::prove_transaction(blockchain, &self.hashes[0])
         }
     }
 }
 
+impl RequestTransactionReceiptsByAddress {
+    const MAX_RECEIPTS: u16 = 500;
```

Comment:
I would say that because the data sent as a response to tx-by-address and receipts-by-address is different (receipts-by-address is much less data per tx), different constants are fine. The work done by the node processing the request is probably the same, though.